universum/lib/utils.py

```diff
@@ -68,7 +76,15 @@ def detect_environment():
     return "terminal"
 
 
-def create_driver(local_factory, teamcity_factory, jenkins_factory, env_type=""):
+T1 = TypeVar('T1')
```

Comment:
please feel free to suggest better naming here

Reply:
T1, T2 and T3 are any Modules.
However, they are definitely for local environment, teamcity environment and jenkins environment.

Reply:
Jenkins factory can either be for JenkinsServerForHostingBuild or for JenkinsServerForTrigger. I am fine with hardcoding acceptable factory types here, but it seems excessive to me. E.g. the way it's written right now we can add other factories for local/teamcity/jenkins without changing this util function.

Reply:
I was only talking about TypeVar names.
They are factories for known environment, so you can use more specific names instead of T1, T2 and T3.
Also, you can bind them to Module.